dotCMS/src/main/java/com/dotcms/visitor/business/VisitorAPIImpl.java

```diff
@@ -105,6 +111,24 @@ public VisitorAPIImpl(LanguageWebAPI languageWebAPI, PersonaAPI personaAPI) {
         return visitorOpt;
     }
 
+    private Persona getPersona(PageMode pageMode, User user, String personaID)
+            throws DotDataException, DotSecurityException {
+        Persona persona = null;
+
+        if (pageMode.showLive) {
+            persona = personaAPI.findLive(personaID, user, true);
+        } else {
+            persona = personaAPI.find(personaID, user, true);
+        }
+
+        if(persona==null) {
+            persona = personaAPI.findPersonaByTag(personaID, user, true)
+                    .orElse(null);
+        }
+
+        return persona;
+    }
+
     public void removeVisitor(final HttpServletRequest request){
         DotPreconditions.checkNotNull(request, IllegalArgumentException.class, "Null Request");
         final HttpSession session = request.getSession(false);
```

Comment:
any test for the change in this class?

Reply:
yes! new graphql test that covers these changes

Reply:
Integration tests also added